### PR TITLE
Fix ZeroDivisionError on html_table.

### LINKED ISSUE #424: ZeroDivisionError in jupyter repr
**Describe the bug**
In jupyter, with a size=0 sparse array, I got an error in the repr

```python-traceback

---------------------------------------------------------------------------
ZeroDivisionError                         Traceback (most recent call last)
~/conda/envs/dask-upstream-dev/lib/python3.8/site-packages/IPython/core/formatters.py in __call__(self, obj)
    343             method = get_real_method(obj, self.print_method)
    344             if method is not None:
--> 345                 return method()
    346             return None
    347         else:

~/conda/envs/dask-upstream-dev/lib/python3.8/site-packages/sparse/_sparse_array.py in _repr_html_(self)
    170         Renders in Jupyter.
    171         """
--> 172         return html_table(self)
    173 
    174     @abstractmethod

~/conda/envs/dask-upstream-dev/lib/python3.8/site-packages/sparse/_utils.py in html_table(arr)
    319         str(arr.shape),
    320         str(arr.nnz),
--> 321         str(arr.nnz / arr.size),
    322     ]
    323 

ZeroDivisionError: division by zero
```

**To Reproduce**
```python
import numpy as np
import sparse

sparse.COO.from_numpy(np.array(()))
```

**System**
 - `sparse` version (`sparse.__version__`) = `0.11.2+12.g440809f`

**Additional context**
This came up when testing dask since the `_meta` is an empty array.

### PR DESCRIPTION
**Pull request by H4R5H1T-007**
Fix #424

### CODE REVIEW

File: sparse/_utils.py [hunk 1]
```diff
@@ -327,26 +327,37 @@ def html_table(arr):
     table = "<table>"
     table += "<tbody>"
     headings = ["Format", "Data Type", "Shape", "nnz", "Density", "Read-only"]
+    try:
+        density = (arr.nnz / arr.size)
+    except ZeroDivisionError:
+        density = 'undefined'
```
Comment (hameerabbasi) [inline sparse/_utils.py:333]:
```suggestion
        density = np.Nan
```

File: sparse/_utils.py [hunk 2]
```diff
@@ -327,26 +327,37 @@ def html_table(arr):
...
+        str(density),
     ]
 
     # read-only
     info.append(str(not hasattr(arr, "__setitem__")))
 
+    try:
+        storage_ratio = "{:.1f}".format(arr.nbytes / (reduce(operator.mul, arr.shape, 1) * arr.dtype.itemsize))
+    except ZeroDivisionError:
+        storage_ratio = 'undefined'
```
Comment (hameerabbasi) [inline sparse/_utils.py:349]:
```suggestion
        storage_ratio = np.man
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Maybe cast these things to `float` so things happen automatically, without a try-catch?

File: sparse/tests/test_coo.py [hunk 3]
```diff
@@ -1606,3 +1606,10 @@ def test_coo_valerr():
 def test_random_idx_dtype():
     with pytest.raises(ValueError):
         sparse.random((300,), density=0.1, format="coo", idx_dtype=np.int8)
+
+
+def test_html_for_size_zero():
+    try:
+        html_table(sparse.COO.from_numpy(np.array(())))
```
Comment (hameerabbasi) [inline sparse/tests/test_coo.py:1613]: The try catch is unnecessary, the test fails on an exception.

Also, we should check the result of density and storage ratio, that's where the original bug was.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
The test needs fixing.

**Comment by H4R5H1T-007** [comment]
Tests are failing in Windows environment @hameerabbasi sir. can you check what is error because I am unable to understand it.

File: sparse/tests/test_coo.py [hunk 4]
```diff
@@ -1606,3 +1607,43 @@ def test_coo_valerr():
 def test_random_idx_dtype():
     with pytest.raises(ValueError):
         sparse.random((300,), density=0.1, format="coo", idx_dtype=np.int8)
+
+
+def test_html_for_size_zero():
+    arr = sparse.COO.from_numpy(np.array(()))
+    ground_truth = "<table>"
```
Comment (hameerabbasi) [inline sparse/tests/test_coo.py:1614]: It'd be ideal if ground truth was a literal string, and you didn't have to construct it here.

Copy-pasting code from the function into the test is usually Bad.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]

**Comment by hameerabbasi** [comment]
Thanks for your work on this, @H4R5H1T-007!

**Comment by H4R5H1T-007** [comment]
Thank You @hameerabbasi sir for your guidance to solve this issue. It helped me to learn a lot.

**Pull request merged**